src/graphql_resolvers.rs

```diff
@@ -1,11 +1,25 @@
-use chrono::NaiveDate;
-use juniper::{graphql_object, EmptyMutation, EmptySubscription, FieldResult, GraphQLObject};
+use chrono::{NaiveDate, NaiveDateTime};
+use juniper::{graphql_object, EmptySubscription, FieldResult, GraphQLObject};
 use uuid::Uuid;
 
+use crate::database;
+use crate::models;
 use crate::services;
 
+// A Cashtools user.
+#[derive(GraphQLObject, Clone)]
+pub struct User {
+    id: Uuid,
+    username: String,
+    register_date: Option<NaiveDateTime>,
+    email: String,
+    last_code_gen_request: Option<NaiveDateTime>,
```

Comment:
I think that every Cash Tools user should have a register date and using an option here is making invalid state representable. We should use other structs to represent other user states.
Relevant resource: https://blog.janestreet.com/effective-ml-revisited/

Reply:
done